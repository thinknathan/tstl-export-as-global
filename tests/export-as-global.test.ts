import * as tstl from "typescript-to-lua";
import path from "path";
import { readFileSync } from "fs";

it("assigns globals", async () => {
	const result = tstl.transpileProject(path.join(__dirname, "tests.tsconfig.json"));
	expect(result.diagnostics).toHaveLength(0);


	const lua = readFileSync(path.join(__dirname, "out", "tests", "test.script.lua"), "utf-8");
    expect(lua).toMatchSnapshot();
});

it("works with modules", async () => {
	const result = tstl.transpileProject(path.join(__dirname, "tests.tsconfig.json"));
	expect(result.diagnostics).toHaveLength(0);


	const lua = readFileSync(path.join(__dirname, "out", "tests", "test-module.lua"), "utf-8");
    expect(lua).toMatchSnapshot();
});

it("handles organic returns", async () => {
	const result = tstl.transpileProject(path.join(__dirname, "tests.tsconfig.json"));
	expect(result.diagnostics).toHaveLength(0);


	const lua = readFileSync(path.join(__dirname, "out", "tests", "test-return.script.lua"), "utf-8");
    expect(lua).toMatchSnapshot();
});

it("handles variables", async () => {
	const result = tstl.transpileProject(path.join(__dirname, "tests.tsconfig.json"));
	expect(result.diagnostics).toHaveLength(0);


	const lua = readFileSync(path.join(__dirname, "out", "tests", "test-var.script.lua"), "utf-8");
    expect(lua).toMatchSnapshot();
});