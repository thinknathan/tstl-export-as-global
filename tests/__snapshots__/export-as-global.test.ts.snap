// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assigns globals 1`] = `
"--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
local ____exports = {}
local function no_export(self)
    local ____ = \\"not-exported\\"
end
function ____exports.init(self)
    no_export(nil)
    local ____ = \\"init\\"
end
function ____exports.update(self, dt)
    local val = dt * 100
    local ____ = val * val
end
function ____exports.exported(self)
    local ____ = \\"exported\\"
end
init = ____exports.init
update = ____exports.update

"
`;

exports[`handles organic returns 1`] = `
"--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
local ____exports = {}
local function try_return(self)
    return \\"not-exported\\"
end
function ____exports.init(self)
    if true then
        return try_return(nil)
    end
end
function ____exports.update(self, dt)
    local val = dt * 100
    local ____ = val * val
end
function ____exports.exported(self)
    local ____ = \\"exported\\"
end
init = ____exports.init
update = ____exports.update

"
`;

exports[`handles variables 1`] = `
"--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
local ____exports = {}
local who = \\"BAR\\"
____exports.what = \\"FOO\\"
____exports.where = 1000
____exports.when = function() return 2000 end
____exports.dontMakeThisGlobal = 3000
____exports.who = who
what = ____exports.what
where = ____exports.where
when = ____exports.when
who = ____exports.who

"
`;

exports[`works with modules 1`] = `
"--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
-- Lua Library inline imports
function __TS__Class(self)
    local c = {prototype = {}}
    c.prototype.__index = c.prototype
    c.prototype.constructor = c
    return c
end

local ____exports = {}
____exports.Template = __TS__Class()
local Template = ____exports.Template
Template.name = \\"Template\\"
function Template.prototype.____constructor(self)
    self.prop = {\\"a prop\\"}
end
function Template.prototype.props(self)
    return self.prop
end
return ____exports
"
`;
